src/Symfony/Component/Workflow/Event/GuardEvent.php

```diff
@@ -11,21 +11,52 @@
 
 namespace Symfony\Component\Workflow\Event;
 
+use Symfony\Component\Workflow\Marking;
+use Symfony\Component\Workflow\Transition;
+use Symfony\Component\Workflow\TransitionBlocker;
+use Symfony\Component\Workflow\TransitionBlockerList;
+
 /**
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Grégoire Pineau <lyrixx@lyrixx.info>
  */
 class GuardEvent extends Event
 {
-    private $blocked = false;
+    private $transitionBlockerList;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($subject, Marking $marking, Transition $transition, $workflowName = 'unnamed')
+    {
+        parent::__construct($subject, $marking, $transition, $workflowName);
+
+        $this->transitionBlockerList = new TransitionBlockerList();
+    }
+
+    public function isBlocked(): bool
+    {
+        return 0 !== count($this->transitionBlockerList);
+    }
+
+    public function setBlocked(bool $blocked): void
+    {
+        if (!$blocked) {
+            $this->transitionBlockerList = new TransitionBlockerList();
+
+            return;
+        }
+
+        $this->transitionBlockerList->add(TransitionBlocker::createUnknownReason($this->getTransition()->getName()));
+    }
 
-    public function isBlocked()
+    public function getTransitionBlockerList(): TransitionBlockerList
```

Comment:
Changing the order of method make the diff harder to read and to review. Could you keep the same method order ?

Thanks

Reply:
The order of methods didn't change. Github's diff tool just struggles to find the similarity, due to the amount of changes in this file. Screenshot of PhpStorm's diff tool:

![image](https://user-images.githubusercontent.com/4347095/35484635-72a0f7d0-044a-11e8-97a1-b9d0e898caca.png)

Let me know, if I missed something.